integrations/clickup/clickup/client.py

```diff
@@ -0,0 +1,116 @@
+        logger.info("Getting teams from Clickup")
+        url = f"{self.clickup_url}/team"
+        team_response = await self.client.get(url)
+        team_response.raise_for_status()
+        team_list = team_response.json()["teams"]
+        yield team_list
+    
+
+
+    async def get_spaces(self)-> AsyncGenerator[list[dict[str, Any]], None]:
+        logger.info("Getting spaces from Clickup using team_ids from get_teams")
+        async for team_list in self.get_teams():
+            for team_dict in team_list:
+                team_id = team_dict["id"]
+                url = f"{self.clickup_url}/team/{team_id}/space"
+                space_response = await self.client.get(url)
+                space_response.raise_for_status()
+                space_list = space_response.json()["spaces"]
+                # record the present team id to be useful for relation in get_projects() function
+                event.attributes[CacheKeys.TEAM]=team_id
+                yield space_list
+
+
+    async def get_folders(self)->AsyncGenerator[list[dict[str, Any]], None]:
+        logger.info("Getting folders from Clickup using space_ids from get_spaces")
+        async for space_list in self.get_spaces():
+            for space_dict in space_list:
+                space_id = space_dict["id"]
+                url = f"{self.clickup_url}/space/{space_id}/folder"
+                folder_response = await self.client.get(url)
+                folder_response.raise_for_status()
+                folder_list = folder_response.json()["folders"]
+                yield folder_list
+
+
+    async def get_projects(self)->AsyncGenerator[list[dict[str, Any]], None]:
+        logger.info("Getting projects from Clickup")
+        async for folder_list in self.get_folders():
+            for folder_dict in folder_list:
+                folder_id = folder_dict["id"]
+                url = f"{self.clickup_url}/folder/{folder_id}/list"
+                project_response = await self.client.get(url)
+                project_response.raise_for_status()
+                project_list = project_response.json()["lists"]
+                team_id = str(event.attributes.get(CacheKeys.TEAM))
+                project_list_with_team = [{**project, "__team": team_id} for project in project_list]
+                yield project_list_with_team
+            
+            async for space_list in self.get_spaces():
+                for space_dict in space_list:
+                    space_id = space_dict["id"]
+                    url = f"{self.clickup_url}/space/{space_id}/list"
+                    project_response = await self.client.get(url)
+                    project_response.raise_for_status()
+                    project_list = project_response.json()["lists"]
+                    team_id = str(event.attributes.get(CacheKeys.TEAM))
+                    project_list_with_team = [{**project, "__team": team_id} for project in project_list]
+                    yield project_list_with_team
+
```

Comment:
Why are you getting spaces within a folder? Isn't it the opposite? You get space_id, you use it to get folders and you use folders to get list